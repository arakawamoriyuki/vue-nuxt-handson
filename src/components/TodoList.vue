<template>
  <div>
    <v-card max-width="800" class="mx-auto" outlined>
      <v-toolbar color="primary" dark flat>
        <v-toolbar-title>Create Todo</v-toolbar-title>
      </v-toolbar>
      <v-card-text>
        <v-form>
          <v-text-field label="title" v-model="newTask.title" />
          <v-text-field label="description" v-model="newTask.description" />
        </v-form>
      </v-card-text>
      <v-card-actions>
        <v-spacer></v-spacer>
        <v-btn color="primary" @click="addTask(newTask)">Create</v-btn>
      </v-card-actions>
    </v-card>
    <v-card max-width="800" class="mx-auto" outlined>
      <v-list subheader two-line flat>
        <v-subheader>Todo</v-subheader>
        <v-list-item-group multiple>
          <v-list-item v-for="task in todoTasks" :key="task.id">
            <v-list-item-action>
              <v-checkbox
                v-model="task.done"
                color="primary"
                @click="done(task.id)"
              ></v-checkbox>
            </v-list-item-action>
            <v-list-item-content>
              <v-list-item-title>{{task.id}}. {{task.title}}</v-list-item-title>
              <v-list-item-subtitle>{{task.description}}</v-list-item-subtitle>
            </v-list-item-content>
            <v-list-item-action>
              <v-icon color="red" @click="removeTask(task.id)">mdi-delete</v-icon>
            </v-list-item-action>
          </v-list-item>
        </v-list-item-group>
      </v-list>
      <v-divider></v-divider>
      <v-list subheader two-line flat>
        <v-subheader>Done</v-subheader>
        <v-list-item-group multiple>
          <v-list-item v-for="task in doneTasks" :key="task.id">
            <v-list-item-action>
              <v-checkbox
                v-model="task.done"
                color="primary"
                @click="retodo(task.id)"
              ></v-checkbox>
            </v-list-item-action>
            <v-list-item-content>
              <v-list-item-title>{{task.id}}. {{task.title}}</v-list-item-title>
              <v-list-item-subtitle>{{task.description}}</v-list-item-subtitle>
            </v-list-item-content>
            <v-list-item-action>
              <v-icon color="red" @click="removeTask(task.id)">mdi-delete</v-icon>
            </v-list-item-action>
          </v-list-item>
        </v-list-item-group>
      </v-list>
    </v-card>
  </div>
</template>

<script>
import store from '@/store/TodoListStore';

export default {
  name: 'todo-list',
  store,
  computed: {
    newTask() {
      return this.$store.state.newTask;
    },
    todoTasks() {
      return this.$store.getters.todoTasks;
    },
    doneTasks() {
      return this.$store.getters.doneTasks;
    },
  },
  methods: {
    addTask(task) {
      this.$store.commit('addTask', task);
    },
    removeTask(taskId) {
      this.$store.commit('removeTask', taskId);
    },
    done(taskId) {
      this.$store.commit('done', taskId);
    },
    retodo(taskId) {
      this.$store.commit('retodo', taskId);
    },
  },
};
</script>
