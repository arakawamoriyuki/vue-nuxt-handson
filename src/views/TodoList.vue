<template>
  <v-responsive
    class="overflow-y-auto"
    max-height="400"
  >
    <div class="pa-6 text-center">
      This is a todo-list page
    </div>
  </v-responsive>
</template>

<script>
export default {
  name: 'todo-list',
};
</script>
