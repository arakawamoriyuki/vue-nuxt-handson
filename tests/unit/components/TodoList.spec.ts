import { shallowMount } from '@vue/test-utils';
import TodoList from '@/components/TodoList.vue';

describe('TodoList', () => {
  it('TodoListが表示される', () => {
    const wrapper = shallowMount(TodoList);
    expect(wrapper).toMatchSnapshot();
  });
});
