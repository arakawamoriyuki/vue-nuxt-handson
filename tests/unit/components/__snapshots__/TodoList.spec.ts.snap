// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList TodoListが表示される 1`] = `
<div>
  <v-card-stub loaderheight="4" tag="div" maxwidth="800" outlined="true" class="mx-auto">
    <todo-form-stub onsubmit="function (newTask) { _vm.addTask(newTask) }"></todo-form-stub>
  </v-card-stub>
  <v-card-stub loaderheight="4" tag="div" maxwidth="800" outlined="true" class="mx-auto">
    <check-list-stub title="Todo" items="[object Object],[object Object],[object Object]" oncheck="function (task) { _vm.done(task.id) }" ondelete="function (task) { _vm.removeTask(task.id) }"></check-list-stub>
    <v-divider-stub></v-divider-stub>
    <check-list-stub title="Done" items="[object Object],[object Object]" oncheck="function (task) { _vm.retodo(task.id) }" ondelete="function (task) { _vm.removeTask(task.id) }"></check-list-stub>
  </v-card-stub>
</div>
`;
